src/nrniv/prcellstate.cpp

```diff
@@ -100,8 +98,6 @@ static void pr_netcon(NrnThread& nt, FILE* f) {
 
 static void pr_realcell(PreSyn& ps, NrnThread& nt, FILE* f) {
     // for associating NetCons with Point_process identifiers
```

Comment:
Perhaps pt2index should be cleared . In the original version there might be a memory leak if  the hoc_execerror fires.
Here, there might be old items when prcellstate is called again. Alternatively, we could clean up just before the hoc_execerror

Reply:
It seems that pnt2index is only used in pr_reallcell and in pr_memb that is called by this function.
Do you think, I can make the map local to this function?

Reply:
I'm not sure that would avoid the memory leak. It's kind of an experimental question.  hoc_execerror does not nicely pop the stack but just resets it when it does a longjump back to its recovery point (e.g. when python called into hoc). We've kind of had this problem with #1457. Anyway, it seems a ```pnt2index.clear()``` before the hoc_execerror is one possible work-around. And near the beginning of prcellstate would certainly guarantee it starts out empty. Without a test, how well can we rely on our logical analysis...   Perhaps a local map would be best (and pass it as arg to needed function calls) But it would still need a clear just before the hoc_execerror.

Reply:
It seems i resolve the problem of `hox_execerror`

Reply:
Ah yes, the pnt2index does not exist yet when it is possible to call hoc_execerror.